config/initializers/delayed_job.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Delayed::ActiveJobPlugin < Delayed::Plugin
+  callbacks do |lifecycle|
+    lifecycle.before(:invoke_job) do |job|
+      job.payload_object.job_data['provider_job_id'] = job.id
+    end
+  end
+end
+
+Delayed::Worker.plugins << Delayed::ActiveJobPlugin
```

Comment:
please move to this initializer this config
Delayed::Worker.destroy_failed_jobs = false
from config/initializers/yeti.rb